### PR TITLE
Fix GenerateHostCerts http fallback with LegacyCerts.

### PR DESCRIPTION
**Pull request by Joerger**
https://github.com/gravitational/teleport/pull/8193 was missing fallback mechanisms for some interactions between old and new server/clients.

Changes:
 - The existing http endpoints `/webapi/host/credentials` and `/:version/server/credentials` will return `LegacyCerts` (formerly named `PackedKeys`).
  - `LegacyCerts` are expected on the clientside in `weblogic` logic and `httpfallback` logic.

I ran into this issue when trying to register a 7.2.0 kube-agent on a 8.0.0-dev (master) auth server.

```
2021-10-04T15:02:42-07:00 DEBU [AUTH]      Registration via proxy server failed. error:[
ERROR REPORT:
Original Error: *trace.BadParameterError Cert: missing parameter
Stack Trace:
	/home/bjoerger/gravitational/teleport/lib/auth/init.go:1131 github.com/gravitational/teleport/lib/auth.ReadSSHIdentityFromKeyPair
	/home/bjoerger/gravitational/teleport/lib/auth/init.go:1056 github.com/gravitational/teleport/lib/auth.ReadIdentityFromKeyPair
	/home/bjoerger/gravitational/teleport/lib/auth/register.go:192 github.com/gravitational/teleport/lib/auth.registerThroughProxy
	/home/bjoerger/gravitational/teleport/lib/auth/register.go:146 github.com/gravitational/teleport/lib/auth.Register
	/home/bjoerger/gravitational/teleport/lib/service/connect.go:363 github.com/gravitational/teleport/lib/service.(*TeleportProcess).firstTimeConnect
	/home/bjoerger/gravitational/teleport/lib/service/connect.go:103 github.com/gravitational/teleport/lib/service.(*TeleportProcess).connect
	/home/bjoerger/gravitational/teleport/lib/service/connect.go:84 github.com/gravitational/teleport/lib/service.(*TeleportProcess).connectToAuthService
	/home/bjoerger/gravitational/teleport/lib/service/connect.go:52 github.com/gravitational/teleport/lib/service.(*TeleportProcess).reconnectToAuthService
	/home/bjoerger/gravitational/teleport/lib/service/service.go:1901 github.com/gravitational/teleport/lib/service.(*TeleportProcess).registerWithAuthServer.func1
	/home/bjoerger/gravitational/teleport/lib/service/supervisor.go:457 github.com/gravitational/teleport/lib/service.(*LocalService).Serve
	/home/bjoerger/gravitational/teleport/lib/service/supervisor.go:246 github.com/gravitational/teleport/lib/service.(*LocalSupervisor).serve.func1
	/usr/local/go/src/runtime/asm_amd64.s:1581 runtime.goexit
User Message: Cert: missing parameter] auth/register.go:149
```

Tested (with local discovery disabled):
 - `v8.0.0-dev` Auth and `v7.0.0` Node 
 - `v7.0.0` Auth and `v8.0.0-dev` Node

### CODE REVIEW

**Comment by Joerger** [comment]
@zmb3 I should have caught this in the review and in prior discussions, so I went ahead and made the fix.

File: lib/auth/register.go [hunk 1]
```diff
@@ -414,3 +415,45 @@ func ReRegister(params ReRegisterParams) (*Identity, error) {
 
 	return ReadIdentityFromKeyPair(params.PrivateKey, certs)
 }
+
+// LegacyCerts is equivalent to proto.Certs, but
+// uses JSON keys expected by old (7.x and earlier) clients
+// DELETE in 9.0.0 (Joerger/zmb3)
+type LegacyCerts struct {
```
Comment (zmb3) [inline lib/auth/register.go:422]: Can we reuse/rename `registerUsingTokenResponseJSON` from #8256?

**Review by zmb3 — APPROVED** [review]
Thanks Brian!

**Review by andrejtokarcik — APPROVED** [review]

**Review by russjones — APPROVED** [review]
Bot.

**Pull request merged**